src/commands/dev/dev.mjs

```diff
@@ -124,9 +125,20 @@ const dev = async (options, command) => {
     startPollingForAPIAuthentication({ api, command, config, site, siteInfo })
   }
 
+  const liveTunnelUrl = await handleLiveTunnel({ options, site, api, settings })
+  const url = liveTunnelUrl || getProxyUrl(settings)
+  process.env.URL = url
```

Comment:
Note for our future selves: it would be nice if we could use [this mechanism](https://github.com/netlify/cli/blob/fe4210037d9e36e28109dea53425b8889adc052b/src/commands/dev/dev.mjs#L92) for setting internal environment variables rather than setting them on `process.env` directly, but that would mean moving the `injectEnvVariables` method further down and we'd need to look into any side-effects of that.

Not worth the hassle for now.